ember-core/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -179,130 +135,52 @@ private[ember] object Parser {
         idx += 1 // Single Advance Every Iteration
       }
 
-      if (throwable != null) ParseHeadersError(throwable)
-      else if (complete)
-        ParseHeadersCompleted(Headers(headers.toList), bv.drop(idx), chunked, contentLength)
-      else
-        ParseHeadersIncomplete(
-          bv,
-          headers.toList,
-          idx,
-          state,
-          Option(name),
-          start,
-          chunked,
-          contentLength)
+      if (throwable != null) {
+        F.raiseError(ParseHeadersError(throwable))
+      } else if (!complete) {
+        F.raiseError(IncompleteHttpMessage())
```

Comment:
Is there any additional context we might provide here?

Reply:
I added in the partially parsed headers. We could go further and produce an error that reflects what the parsing state was, but I'm more inclined to leave that to a parser library once we switch to that